services/api/service.go

```diff
@@ -635,6 +749,38 @@ func (api *RelayAPI) updateProposerDuties(headSlot uint64) {
 	api.log.Infof("proposer duties updated: %s", strings.Join(_duties, ", "))
 }
 
+func (api *RelayAPI) updateOptimisticSlot(headSlot uint64) {
+	// Wait until there are no optimistic blocks being processed. Then we can
+	// safely update the slot.
+	api.optimisticBlocks.Wait()
```

Comment:
if using an atomic `optimisticSlot`, then the wait might not be needed here 🤔

Reply:
i think we still need the wait group, right? `api.optimisticBlocks` is a waitgroup that we increment every time that we start processing a new optimistic block. we need to wait until that wg is empty. am i missing something ?

Reply:
Yeah good point! Maybe this function could have a better name, like `prepareOptimisticBuildersForSlot`?

Reply:
right